Marlin/src/module/planner.cpp

```diff
@@ -799,14 +797,14 @@ void Planner::calculate_trapezoid_for_block(block_t * const block, const_float_t
   uint32_t initial_rate = LROUND(block->nominal_rate * entry_factor),
            final_rate = LROUND(block->nominal_rate * exit_factor); // (steps per second)
 
-  // Legacy check against supposed timer overflow. However Stepper::calc_timer_interval() already
-  // should protect against it. But removing this code produces judder in direction-switching
-  // moves. This is because the current discrete stepping math diverges from physical motion under
-  // constant acceleration when acceleration_steps_per_s2 is large compared to initial/final_rate.
+  // Limit minimal step rate (Otherwise the timer will overflow.)
```

Comment:
The minimal step rate is enforced to match the one in `calc_timer_interval()` (which is the one that actually protects against timer overflow).